tools/mo/openvino/tools/mo/ops/slice_like.py

```diff
@@ -38,10 +38,10 @@ def infer(node):
             assert input_shape.size == shape_like.size,\
                 'Input shape ranks are inconsistent: {} and {}'.format(input_shape.size, shape_like.size)
             node.axes = int64_array(range(shape_like.size))
-            new_shape = np.copy(shape_like)
+            new_shape = shape_like.copy()
         node.out_port(0).data.set_shape(new_shape)
 
-        if input_value is not None:
+        if input_value is not None and is_fully_defined(shape_like):
```

Comment:
please explain why this is changed

Reply:
If we try to do value inference and `shape_like` had dynamic shape then  we will face error in
https://github.com/openvinotoolkit/openvino/blob/master/tools/mo/openvino/tools/mo/ops/slice_like.py#L52-L53

Note that `new_shape` is calculated from `shape_like` and if `shape_like` has dynamic values then `new_shape` has as well. To make it more clean I modified this condition to check if `new_shape` is not dynamic.